src/ufuncs.c

```diff
@@ -2160,7 +2160,7 @@ static void polygonize_func(char** args, const npy_intp* dimensions, const npy_i
 
   GEOS_INIT;
 
-  GEOSGeometry** geoms = malloc(sizeof(void*) * dimensions[1]);
+  const GEOSGeometry** geoms = malloc(sizeof(void*) * dimensions[1]);
```

Comment:
Is it "correct" to mark `geoms` as const if we assign into `geoms` in the loop below to fill the array?

Reply:
Yes. The type is changed from “pointer to pointer to `GEOSGeometry`” to “pointer to pointer to `const GEOSGeometry`”, or conceptually, from “array of pointers to `GEOSGeometry`” to “array of pointers to `const GEOSGeometry`.” We only assign pointers; we don’t modify the `GEOSGeometry` they reference, so this is fine.

We couldn’t change it to `const GEOSGeometry* const*`, because that *would* make the pointers `const`.

Reply:
Ah, I was reading it as const array instead of array of const Geometry*. Thanks for the clarification!